packages/react/src/patterns/sub-patterns/TableOfContents/TableOfContents.js

```diff
@@ -33,34 +31,31 @@ const { prefix } = settings;
  */
 const TableOfContents = ({ menuItems, children, menuLabel, theme }) => {
   const [selectedId, setSelectedId] = useState(menuItems[0].id);
-  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].title);
-
+  const [selectedTitle, setSelectedTitle] = useState(menuItems[0].name);
+  const useMenuItems = menuItems ? [...menuItems] : findMenuItems();
   useEffect(() => {
     scrollStop(setSelectedItem);
   });
 
+  if (!useMenuItems) {
+    findMenuItems();
+  }
```

Comment:
@carthick move this to inside the `useEffect`. Also, as I mentioned, move the method `findMenuItems` itself to outside of `const TableOfContents`, and rename to `_findMenuItems`.

In general, anything that needs to be called on the client side needs to be inside the `useEffect`.